1 getting started readme

helmoski opened the pull request:
- Target Audience
- Prereqs
- Table of Contents

Commit:
What is EF and Who should use this tutorial sections

Commit:
Completed first draft of getting-started page

Commit:
Changed getting-started page to the README of getting-started directory

arynhuck committed:
Editted README to be concise

The README needed to be reorganized (some content will go into the
Introduction section) and a lot of chit chat needed to be cut down. The
tone is less conversational and more down-to-business (but still
friendly). This will close issue #1

arynhuck committed:
Cut SQL comment

We don't want to tie this EF Core tutorial to a certain type of
database, so I've removed SQL as a requirement. This should make this
README ready to :shipit: . #1

arynhuck committed:
Condensed sentence

Yay wordiness. Took Ryan's suggestions and made the general description
more concise.

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,22 @@
+# Getting Started with Entity Framework Core
+
+## Who is this tutorial meant for?
```
The C# and ASP.NET Core tutorials both use the headings "Who Should Read This" and "Topics Covered" for this page, so we probably should too to stay consistent. 

https://github.com/LadyNaggaga/training-tutorials/blob/master/content/csharp/getting-started/README.md
https://github.com/LadyNaggaga/training-tutorials/blob/master/content/asp.net/getting-started/README.md

arynhuck replied:
Done

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,22 @@
+
+You should understand the following concepts before starting this tutorial:�
+
+* C# (see the [interactive C# tutorial](https://www.microsoft.com/net/tutorials/csharp/getting-started))
```
Looking at how the dot.net website works, the initial tutorial page actually isn't drawn from this file; instead, it is a hard-coded string that is placed within a paragraph. Thus, it will be difficult for us to do bullets unless we change the way that part of the site works. Could we list these prereqs in sentences instead? It would also be better if we could add the prereqs to the "Who Should Read This" section because we won't be able to do headings either.

helmoski replied:
Here is an excerpt from the ASP.NET tutorial:

"This tutorial is aimed at people who are learning ASP.NET Core for the first time. Familiarity with C# is assumed. If you're new to C#, you may wish to go through the Getting Started with C# tutorial before beginning this one."

arynhuck replied:
Done.

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,22 @@
+* Relational database concepts
+* LINQ statements (see this [interactive tutorial](https://www.microsoft.com/net/tutorials/csharp/getting-started/linq) for an introduction to LINQ)
+
+## Table of Contents
```
The C# tutorial has a [table of contents file](https://github.com/LadyNaggaga/training-tutorials/blob/master/content/csharp/getting-started/toc.md) named "toc.md" in addition to the table of contents listed in the README.md. Could you add a toc.md? Make sure to use the same formatting because the website parses this file to determine the titles and locations of the tutorial chapters.

arynhuck replied:
Yeah, I'll move this.

arynhuck replied:
Done!

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,22 @@
+
+This tutorial is meant for those new to Entity Framework (EF) Core, but if you already have an understanding of EF Core, consider this a refresher course. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) .
+
+If you know what you're looking for, feel free to browse by topic on the side. If not, welcome! Let's get started together!
```
"If you know what you're looking for, feel free to browse by topic on the side." => I get this, but I think the use of sidebar navigation is commonplace enough that we don't need to state they can browse by topic. It ends up just being more for the user to read.

"If not, welcome! Let's get started together!" => I understand the appeal to saying this, but it doesn't really match the tone of the other tutorials. I'd like the tone to be less like an individual person's blog. That might just be my perception, though.

arynhuck replied:
We didn't want to be friends with the users anyways. That's fine.

But fixed!

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,22 @@
+
+## Who is this tutorial meant for?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, but if you already have an understanding of EF Core, consider this a refresher course. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) .
```
We probably don't need to include the "refresher course" part. Stating that the tutorial is meant for beginners will be enough to give experienced devs an idea of whether this tutorial will be useful to them.

arynhuck committed:
Changed README content, added TOC

The README has been updated to be more concise and to fit with the other
training-tutorials. A Table of contents has been added (when the other
pages are added to the project, it will have to be updated to contain
links to the markdown pages).

arynhuck commented:
Brought in all the changes from the other branch. Should be good to go now.

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,7 @@
+# Getting Started with Entity Framework Core
```
This file should have the table of contents too. Sorry. I know it's duplicate data, but that is how they have it set up.

arynhuck replied:
😐

arynhuck replied:
Is this what you meant:

## Topics Covered

This tutorial will cover the following topics:
- Introduction to Entity Framework Core
- Adding Entity Framework Core to a Project

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,7 @@
+
+This tutorial is meant for those new to Entity Framework (EF) Core, but if you already have an understanding of EF Core, consider this a refresher course. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) .
+
+Before beginning this tutorial, you should understand a few topics. One, this tutorial assumes a familiarity with C#. If you're new to C#, you may wish to go through the [Getting Started with C#](https://www.microsoft.com/net/tutorials/csharp/getting-started) tutorial. Two, you should have a basic knowledge of relational database concepts. Three, this tutorial will use LINQ statements. If you hav ...
```
This may be personal preference, but could we use first, second, third instead of one, two, three?

rdahlman3 replied:
Yeah, that's personal preference. I like "one, two, three" better because the tutorial states that it assumes knowledge of a few basic concepts. Unless @arynhuck or @davidmikh agree with you, I'd say you're outvoted 😛

deakgeek replied:
I prefer first, second, and third if we are using this paragraph, not a list.

abadami replied:
I am a fan of first, second, third, but either way works I think.

helmoski replied:
@rdahlman3 Not outvoted 😛

rdahlman3 replied:
But also, does it really matter that much?

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,7 @@
+
+This tutorial is meant for those new to Entity Framework (EF) Core, but if you already have an understanding of EF Core, consider this a refresher course. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) .
+
+Before beginning this tutorial, you should understand a few topics. One, this tutorial assumes a familiarity with C#. If you're new to C#, you may wish to go through the [Getting Started with C#](https://www.microsoft.com/net/tutorials/csharp/getting-started) tutorial. Two, you should have a basic knowledge of relational database concepts. Three, this tutorial will use LINQ statements. If you hav ...
```
Could you just say "LINQ" instead of "LINQ statements"?

helmoski replied:
"for an introduction to LINQ)" <= I think this may have been left in by accident

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,7 @@
+
+## Who Should Read This?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, but if you already have an understanding of EF Core, consider this a refresher course. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) .
```
Could we remove the "if you already have an understanding..." part? The other tutorials don't say it, and I don't think we need to either. Experienced coders will understand they can use it as a refresher course without us needing to say it since we say it is for beginners.

helmoski replied:
If you think the first paragraph is too short, I would combine the first and second paragraph. That's how the other tutorials have it anyways.

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,7 @@
+
+## Who Should Read This?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, but if you already have an understanding of EF Core, consider this a refresher course. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) .
```
Could you state that this tutorial covers basic functionality or something like that? It will make the "More advanced functionality..." part make more sense.

rdahlman3 committed:
added table of contents section, combined paragraphs, fixed wordingof LINQ section and where we mention more advanced functionality

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,9 @@
+
+## Who Should Read This?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, as it covers the basic functionalities of EF Core. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) . Before beginning this tutorial, you should understand a few topics. One, this tutorial assumes a familiarity with C#. If you're new to C#, you may wish to go through the [Getting Started ...
```
### Prefer first, second, third:

1) Ryan
2) Melia
3) Andrew (said he thinks one, two, three works but would prefer first, second, third)

### Prefer one, two, three:

1) Aryn
2) Rebecca

### Unknown

1) David

arynhuck replied:
Why can't we have lists again?

helmoski replied:
### Prefer Bullets:

Everyone

arynhuck committed:
TOC added to README

Apparently Ryan wanted this.

arynhuck committed:
Changing how the list of prereqs looks

We went with 1) 2) 3)

arynhuck commented:
Made the changes @helmoski 🌮

helmoski commented on content/efcore/getting-started/toc.md:
```diff
@@ -0,0 +1,2 @@
+- Introduction to Entity Framework Core
```
These will be the names of the chapters displayed on the sidebar of the website. To avoid them being too long, let's abbreviate Entity Framework:

Intro to EF Core
Adding EF Core to a Project

helmoski replied:
Please update the names in the README table of contents too.

arynhuck replied:
fixed

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,11 @@
+
+## Who Should Read This?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, as it covers the basic functionalities of EF Core. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) . Before beginning this tutorial, you should understand a few topics. 1) this tutorial assumes a familiarity with C#. If you're new to C#, you may wish to go through the [Getting Started w ...
```
The period after the second sentence has an extra space before it.

arynhuck replied:
fixed

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,11 @@
+
+## Who Should Read This?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, as it covers the basic functionalities of EF Core. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) . Before beginning this tutorial, you should understand a few topics. 1) this tutorial assumes a familiarity with C#. If you're new to C#, you may wish to go through the [Getting Started w ...
```
Could you remove "on LINQ" from the last sentence? Just "can help you get up-to-speed."

arynhuck replied:
fixed

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -0,0 +1,11 @@
+
+## Who Should Read This?
+
+This tutorial is meant for those new to Entity Framework (EF) Core, as it covers the basic functionalities of EF Core. More advanced functionality is covered in the [docs](https://docs.efproject.net/en/latest/) . Before beginning this tutorial, you should understand a few topics. 1) this tutorial assumes a familiarity with C#. If you're new to C#, you may wish to go through the [Getting Started w ...
```
So I just realized that the markdown links aren't going to work either because of the way the site works. Let's just do the markdown the way we want it. We will adjust the hard-coded description to make it work for the site.

In other words, go ahead and use bullets, and still keep the links.

arynhuck replied:
Woo! Bullet points for the win!

arynhuck committed:
Conciseness changes to README, emptying TOC

We shortened the titles of articles, changed back to bullet points (the
markdown supports it, we will just have to change what we give the
website), and changed wordage.
Emptied the TOC so nothing will freak out when it tries to parse
documents that haven't been added yet.

arynhuck commented:
I have made all the changes. 🌮

helmoski commented:
:shipit:
